packages/libs/eda/src/lib/map/analysis/mapTypes/plugins/DonutMarkerMapType.tsx

```diff
@@ -453,9 +485,75 @@ function MapOverlayComponent(props: MapTypeMapLayerProps) {
   const toggleStarredVariable = useToggleStarredVariable(props.analysisState);
   const noDataError = getLegendErrorMessage(data.error);
 
+  const updateLegendPosition = useDebouncedCallback(
+    (position: PanelConfig['position']) => {
+      setLegendPanelConfig({
+        ...legendPanelConfig,
+        pie: {
+          variable: position,
+        },
+      });
+    },
+    250
+  );
+
+  const updateVisualizationPosition = useCallback(
+    (position: PanelConfig['position']) => {
+      const newConfig = {
+        position,
+        dimensions:
+          visualizationPanelConfigs?.pie?.dimensions ??
+          DEFAULT_DRAGGABLE_VIZ_DIMENSIONS,
+        isVisible: true,
+      };
+      setVisualizationPanelConfigs({
+        ...visualizationPanelConfigs,
+        pie: newConfig,
+      });
+    },
+    [visualizationPanelConfigs, setVisualizationPanelConfigs]
+  );
+
+  const updateVisualizationDimensions = useCallback(
+    (dimensions: PanelConfig['dimensions']) => {
+      const newConfig = {
+        dimensions,
+        position:
+          visualizationPanelConfigs?.pie?.position ??
+          DEFAULT_DRAGGABLE_VIZ_POSITION,
+        isVisible: true,
+      };
+      setVisualizationPanelConfigs({
+        ...visualizationPanelConfigs,
+        pie: newConfig,
+      });
+    },
+    [visualizationPanelConfigs, setVisualizationPanelConfigs]
+  );
+
+  const onPanelDismiss = useCallback(() => {
+    setActiveVisualizationId(undefined);
+    setVisualizationPanelConfigs({
+      ...visualizationPanelConfigs,
+      pie: {
+        position:
+          visualizationPanelConfigs?.pie?.position ??
+          DEFAULT_DRAGGABLE_VIZ_POSITION,
+        dimensions:
+          visualizationPanelConfigs?.pie?.dimensions ??
+          DEFAULT_DRAGGABLE_VIZ_DIMENSIONS,
```

Comment:
Can't we just spread the old values?

```suggestion
        ...visualizationPanelConfigs?.pie,
```